go/src/koding/klientctl/commands/autocomplete.go

```diff
@@ -0,0 +1,115 @@
+package commands
+
+const (
+	bashCompletionFunc = `
+# This function adds available remote machines to COMREPLY.
+# Usage: __kd_remote_machines [OPTIONS]
+# Available OPTIONS:
+#     -p PREFIX Prefix to add to each machine name
+#     -s SUFFIX Suffix to add to each machine name
+#
+__kd_remote_machines()
+{
+    local kd_output flag OPTIND=1
+    local words=()
+
+    if kd_output=$(kd machine identifiers 2>/dev/null); then
+        out=($(echo "${kd_output}"))
+        while getopts "p:s:" flag "$@"; do
+            case $flag in
+                p) words=${out[@]/#/$OPTARG}; out=(${words[@]}) ;;
+                s) words=${out[@]/%/$OPTARG}; out=(${words[@]}) ;;
+            esac
+        done
+
+        COMPREPLY=( "${COMPREPLY[@]}" $( compgen -W "${words[*]}" -- "$cur" ) )
+    fi
+}
+
+# This command provides auto completion abilities for exec subcommand.
+__kd_exec_completion()
+{
+    if [[ ${prev} == "exec" ]]; then
```

Comment:
This var is a standard name that holds last token. It's generated by cobra in its main function:

```bash
__start_kd()
{
    local cur prev words cword
    declare -A flaghash 2>/dev/null || :
    if declare -F _init_completion >/dev/null 2>&1; then
        _init_completion -s || return
    else
        __my_init_completion -n "=" || return
    fi

    local c=0
    local flags=()
    local two_word_flags=()
    local local_nonpersistent_flags=()
    local flags_with_completion=()
    local flags_completion=()
    local commands=("kd")
    local must_have_one_flag=()
    local must_have_one_noun=()
    local last_command
    local nouns=()

    __handle_word
}

if [[ $(type -t compopt) = "builtin" ]]; then
    complete -o default -F __start_kd kd
else
    complete -o default -o nospace -F __start_kd kd
fi

```